jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-client/src/main/java/org/jbpm/workbench/cm/client/actions/NewActionViewImpl.java

```diff
@@ -0,0 +1,242 @@
+
+    @Inject
+    @DataField("user-name-input")
+    TextInput userNameInput;
+
+    @Inject
+    @DataField("user-name-help")
+    Span userNameHelp;
+
+    @Inject
+    @DataField("modal")
+    private Modal modal;
+
+    private Boolean forUser;
+
+    private Command okCommand;
+
+    @Inject
+    private TranslationService translationService;
+
+    @PostConstruct
+    public void init() {
+        this.actionNameLabel.addRequiredIndicator();
+        this.roleNameLabel.addRequiredIndicator();
+    }
+
+    @Override
+    public void init(final CaseActionListsPresenter presenter) {
+    }
+
+    @Override
+    public void show(boolean forUser, final Set<String> roles, final Command okCommand) {
+        clearErrorMessages();
+        clearValues();
+
+        this.forUser = forUser;
+
+        userNameLabel.getElement().setTextContent(translationService.format(forUser ? USER : GROUP));
+        userNameLabel.addRequiredIndicator();
+
+        roles.forEach(r -> roleNameList.addOption(r));
+        roleNameList.refresh();
+
+        this.okCommand = okCommand;
+
+        modal.show();
+    }
+
+    @Override
+    public void hide() {
+        modal.hide();
+    }
+
+    private boolean validateForm() {
+        clearErrorMessages();
+
+        final boolean roleNameEmpty = isNullOrEmpty(roleNameList.getValue());
+        final boolean userNameEmpty = isNullOrEmpty(userNameInput.getValue());
+        if (roleNameEmpty && userNameEmpty) {
+            roleNameList.getElement().focus();
+            roleNameHelp.setTextContent(translationService.format(PLEASE_SELECT_ROLE));
+            roleNameGroup.setValidationState(ValidationState.ERROR);
+            userNameInput.focus();
+            userNameHelp.setTextContent(translationService.format(forUser ? USER_REQUIRED : GROUP_REQUIRED));
+            userNameGroup.setValidationState(ValidationState.ERROR);
+            return false;
+        }
+
+        roleNameGroup.setValidationState(ValidationState.SUCCESS);
+        userNameGroup.setValidationState(ValidationState.SUCCESS);
+
+        return true;
+    }
+
+    private void clearValues() {
+        roleNameList.setValue("");
+        roleNameList.removeAllOptions();
+        roleNameList.refresh();
+        userNameInput.setValue("");
+        actionNameInput.setValue("");
+        actionDescInput.setValue("");
+    }
+
+    private void clearErrorMessages() {
+        roleNameHelp.setTextContent("");
+        userNameHelp.setTextContent("");
+        userNameGroup.clearValidationState();
+        roleNameGroup.clearValidationState();
+    }
+
+    @Override
+    public String getTaskName() {
+        return actionNameInput.getValue();
+    }
+
+    @Override
+    public String getDescription() {
+        return actionDescInput.getValue();
+    }
+
+    @Override
+    public String getActors() {
+        return userNameInput.getValue();
+    }
+
+    @Override
+    public String getGroups() {
+        return null;
```

Comment:
Should we add support so that new tasks are assigned to either users and/or groups?